service/worker/scheduler/workflow.go

```diff
@@ -129,7 +126,7 @@ var (
 		FutureActionCountForList:          5,
 		RecentActionCountForList:          5,
 		MaxSearchAttrLen:                  2000, // server default is 2048 but leave a little room
-		IterationsBeforeContinueAsNew:     500,
+		IterationsBeforeContinueAsNew:     501,
```

Comment:
just curious, why?

Reply:
This is to compensate for fixing the bug in line 170: if this value was 10 it would do 11 iterations. So the old behavior was to do 501 iterations. To keep the old behavior, this value should change to 501. Otherwise if an upgrade happened on iteration 500, the new version would continue-as-new early, which would be a non-determinism error. At least I think that's what would happen... this is a little confusing to reason about because of the mutablesideeffect thing.

Reply:
Maybe worth put this explanation as comment.